### ISSUE TITLE
Clearer "starting point" in imgui-examples

### ISSUE THREAD

**Issue by dbr** [imgui-rs/imgui-rs#621]
Split from #619 -

I suspect a lot of new users start, quite reasonably, by looking at `imgui-examples/`. This directs people towards copying `support/mod.rs` to their own projects - which I don't think is the best starting point.

The support module is good as it keeps each example (e.g `radio_button.rs`) short and just contains the code to create a radio button or whatever. However the callback approach makes things a bit hard to follow when trying to make more custom integrations (e.g I know it took me a while to realize simple things like "how do I set the title of the host window")

Basically I think there are two separate types of examples, "API examples" and "integration examples", but currently the `imgui-glium-renderer` examples serve as both.

I think we probably just need to:

1. Add one or two simple, self contained, examples of how to use winit+glium to create a widnow (pretty much exactly as done for the glow renderer)
2. Put a nice prominent comment in the `support/mod.rs` explaining this code serves to show how to create "imgui content", and direct people towards these self-contained examples for their own integrations.
3. Somewhat tangentally, maybe switch the imgui-examples to use `imgui-glow-example`? :shrug:

**Comment by WesleyAC** [comment]
I just ran into this as well — I'm trying to figure out how to use winit and glium, since that's what the docs say to do, but I can't find any example of how to do so.

This issue makes it sound like there is an example somewhere — where could I find that? The `imgui-glium-renderer` directory doesn't seem to have any examples in it.

**Comment by dbr** [comment]
> This issue makes it sound like there is an example somewhere — where could I find that? The imgui-glium-renderer directory doesn't seem to have any examples in it.

There's no nice self-contained example of how how to use the glium-renderer - fixing that that is the aim of this issue!

However the `imgui-examples` directory uses glium, and the main glium+winit integration is here:

https://github.com/imgui-rs/imgui-rs/blob/v0.8.2/imgui-examples/examples/support/mod.rs

They just aren't as nicely contained as the examples for glow, which can be found here `glow_01_basic.rs`:

https://github.com/imgui-rs/imgui-rs/tree/v0.8.2/imgui-glow-renderer/examples

If you are starting with the `support/mod.rs` for an application, I'd recommend refactoring/simplfying it it as much as you see fit. Things I found helpful were:

- removing the `System` struct and instead having just `init` method, makes it much easier
- replacing the `&mut bool` for quitting (in the `run_ui` closure) with a simple message queue, basically `Vec<WindowMessages>` and `enum WindowMessage { Quit, SetWindowTitle(String), ... }` (before I realized this, I was trying to pass various things in to the closure and it was either messy or impossible)

**Comment by dbr** [comment]
There's now an `examples` folder for all of the main crates in the project, the main addition being a new example for glium consistent with `glow_01_basic`
